src/Projects/Server/Cida.Server/Infrastructure/FtpClient.cs

```diff
@@ -69,18 +68,55 @@ public async Task SaveFileAsync(byte[] file, params string[] path)
             request.Method = WebRequestMethods.Ftp.UploadFile;
 
             await using var stream = await request.GetRequestStreamAsync();
-            await stream.WriteAsync(file, 0 , file.Length);
+            await stream.WriteAsync(file, 0, file.Length);
             using var response = await request.GetResponseAsync();
             this.logger.Info("Uploaded file: {value1}", path);
         }
 
         private FtpWebRequest CreateRequest(params string[] path)
         {
             const string separator = "/";
-            var result = (FtpWebRequest)WebRequest.Create($"ftp://{this.settings.Host}:{this.settings.Port}{separator}{string.Join(separator, path)}");
+            var result = (FtpWebRequest) WebRequest.Create(
+                $"ftp://{this.settings.Host}:{this.settings.Port}{separator}{string.Join(separator, path)}");
             result.Credentials = new NetworkCredential(this.settings.Username, this.settings.Password);
             return result;
         }
+
+        public bool ValidateConfiguration(ExternalServerConnection ftpConnection)
+        {
+            if (string.IsNullOrEmpty(ftpConnection.Host) || string.IsNullOrEmpty(ftpConnection.Username))
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        public bool TryConnect(ExternalServerConnection ftpConnection, out Exception occuredException)
+        {
+            try
+            {
+                var result = (FtpWebRequest) WebRequest.Create($"ftp://{ftpConnection.Host}:{ftpConnection.Port}");
+                result.Credentials = new NetworkCredential(ftpConnection.Username, ftpConnection.Password);
+                result.Method = WebRequestMethods.Ftp.ListDirectory;
+                using var response = (FtpWebResponse) result.GetResponse();
+
+                if (response.StatusCode != FtpStatusCode.OpeningData)
```

Comment:
Fails if FTP connection is already open and usable and does not need to be created first (FTP status code "DataAlreadyOpen").
The connection is already open and usable and thus the test should not fail then!
```suggestion
                if (response.StatusCode != FtpStatusCode.OpeningData && response.StatusCode != FtpStatusCode.DataAlreadyOpen)
```